src/Microsoft.ML.LightGbm/LightGbmArguments.cs

```diff
@@ -1,678 +1,351 @@
+            /// For example, if you set it to 0.8, LightGBM will select 80% of features before training each tree.
+            /// This can be used to speed up training and to reduce over-fitting. Valid range is (0,1].
+            /// </value>
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "Subsample ratio of columns when constructing each tree. Range: (0,1].",
+                ShortName = "ff")]
+            [TlcModule.Range(Inf = 0.0, Max = 1.0)]
+            public double FeatureFraction = 1;
 
-        /// <summary>
-        /// DART booster (Dropouts meet Multiple Additive Regression Trees)
-        /// </summary>
-        /// <remarks>
-        /// For details, please see <a href="https://arxiv.org/abs/1505.01866">here</a>.
-        /// </remarks>
-        public sealed class DartBooster : BoosterParameter<DartBooster.Options>
-        {
-            internal const string Name = "dart";
-            internal const string FriendlyName = "Tree Dropout Tree Booster";
+            /// <summary>
+            /// The L2 regularization term on weights.
+            /// </summary>
+            /// <value>
+            /// Increasing this value could help reduce over-fitting.
+            /// </value>
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "L2 regularization term on weights, increasing this value will make model more conservative.",
+                ShortName = "l2")]
+            [TlcModule.Range(Min = 0.0)]
+            [TGUI(Label = "Lambda(L2)", SuggestedSweeps = "0,0.5,1")]
+            [TlcModule.SweepableDiscreteParam("RegLambda", new object[] { 0f, 0.5f, 1f })]
+            public double L2Regularization = 0.01;
 
             /// <summary>
-            /// The options for <see cref="DartBooster"/>, used for setting <see cref="Booster"/>.
+            /// The L1 regularization term on weights.
             /// </summary>
-            [TlcModule.Component(Name = Name, FriendlyName = FriendlyName, Desc = "Dropouts meet Multiple Additive Regression Trees. See https://arxiv.org/abs/1505.01866")]
-            public sealed class Options : TreeBooster.Options
+            /// <value>
+            /// Increasing this value could help reduce over-fitting.
+            /// </value>
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "L1 regularization term on weights, increase this value will make model more conservative.",
+                ShortName = "l1")]
+            [TlcModule.Range(Min = 0.0)]
+            [TGUI(Label = "Alpha(L1)", SuggestedSweeps = "0,0.5,1")]
+            [TlcModule.SweepableDiscreteParam("RegAlpha", new object[] { 0f, 0.5f, 1f })]
+            public double L1Regularization = 0;
+
+            BoosterParameterBase IComponentFactory<BoosterParameterBase>.CreateComponent(IHostEnvironment env)
             {
```

Comment:
>{ [](start = 12, length = 1)

nit: you can just `BoosterParameterBase IComponentFactory<BoosterParameterBase>.CreateComponent(IHostEnvironment env) => BuildOptions();`